### PR TITLE
Sr system user fk client id refactor

### PR DESCRIPTION
**Pull request by simen-rekkedal**
removed the client_id from System_User table
modify externalId lookup (clientId) to still work, by resolving clientId to system_id

## Related Issue(s)
- #{issue number}

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

## Documentation
- [x] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/Authentication/Services/SystemUserService.cs [hunk 1]
```diff
@@ -49,8 +49,7 @@ public SystemUserService(ISystemUserRepository systemUserRepository, ISystemRegi
             {                
                 IntegrationTitle = request.IntegrationTitle,
                 ProductName = request.ProductName,
```
Comment (acn-dgopa) [inline src/Authentication/Services/SystemUserService.cs:51]: I suggest that we use the same PR to change the model names. Remi had raised a concern about different names for the same data in different places. To be consistent, we will remove the word product from our models. We call it systemname, systemid and not productname/product id. Also are we referring the supplier to the systemvendor or apivendor?

Comment (simen-rekkedal) [reply src/Authentication/Services/SystemUserService.cs:51]: ok

File: src/Persistance/Migration/v0.01/01-alter-table-systemregister.sql [hunk 2]
```diff
@@ -11,4 +11,5 @@ ALTER TABLE altinn_authentication_integration.system_register
 RENAME COLUMN default_rights to rights;
 
 ALTER TABLE altinn_authentication_integration.system_register
-RENAME COLUMN friendly_product_name to system_name;
\ No newline at end of file
+RENAME COLUMN friendly_product_name to system_name;
```
Comment (acn-dgopa) [inline src/Persistance/Migration/v0.01/01-alter-table-systemregister.sql:14]: what has changed here_

File: src/Core/Models/RegisterSystemResponse.cs [hunk 3]
```diff
@@ -14,15 +14,30 @@ namespace Altinn.Platform.Authentication.Core.SystemRegister.Models
     public class RegisterSystemResponse
     {
         /// <summary>
-        /// The unique Id for this product, in human-readable string format.
-        /// The id is in the format of system_vendor_name_plus_name_chosen_by_them.
+        /// The primary key to store and edit the Registered System.
+        /// Used as a Foreign Key in the System User table.
```
Comment (acn-dgopa) [inline src/Core/Models/RegisterSystemResponse.cs:18]: I have changed that as well in my new PR :)

File: src/Core/Models/SystemUserRequestDTO.cs [hunk 4]
```diff
@@ -13,7 +13,7 @@ namespace Altinn.Platform.Authentication.Core.Models
     public class SystemUserRequestDto
     {
         [AllowNull]
-        [JsonPropertyName("PartyId")]
+        [JsonPropertyName("partyId")]
```
Comment (acn-dgopa) [inline src/Core/Models/SystemUserRequestDTO.cs:16]: i remember we had a discussion about saving both partyid and organization number of the org that creates the system user. I don't see the field addition and property for the same here

Comment (simen-rekkedal) [reply src/Core/Models/SystemUserRequestDTO.cs:16]: That is right, will add it.

File: src/Core/Models/RegisterSystemResponse.cs (hunk 3 above)
Comment (simen-rekkedal) [reply src/Core/Models/RegisterSystemResponse.cs:18]: updated from your branch

File: src/Core/Models/SystemUser.cs [hunk 5]
```diff
@@ -21,29 +21,44 @@ public class SystemUser
...
         /// </summary>
-        [JsonPropertyName("ownedByPartyId")]
-        public string OwnedByPartyId { get; set; } = string.Empty;
+        [JsonPropertyName("partyId")]
+        public string PartyId { get; set; } = string.Empty;
+
+        /// <summary>
+        /// The Organisation Number for the end-user as it is stored in ER Registry        
+        /// </summary>
+        [JsonPropertyName("partyOrgNo")]
```
[the remark this answers is not in the document]
Comment (simen-rekkedal) [reply src/Core/Models/SystemUser.cs]: will change to Reportee in new commit

File: src/Core/Models/SystemUserRequestDTO.cs [hunk 6]
```diff
@@ -13,17 +13,25 @@ namespace Altinn.Platform.Authentication.Core.Models
...
+        public string PartyOrgNo { get; set; } 
+
         /// <summary>
-        /// The Title and Description are strings set by the end-user in the Frontend.
+        /// The Title is set by the end-user in the Frontend, by default it is the same as the System's Display Name
+        /// Even if this DTO allows null, the db field is of course still required   
         /// </summary>
         [AllowNull]
-        [JsonPropertyName("IntegrationTitle")]
+        [JsonPropertyName("integrationTitle")]
```
[the remark this answers is not in the document]
Comment (simen-rekkedal) [reply src/Core/Models/SystemUserRequestDTO.cs:30]: removed

File: src/Core/Models/SystemUserUpdateDTO.cs [hunk 7]
```diff
@@ -19,24 +19,24 @@ public class SystemUserUpdateDto
         /// Even if this DTO allows null, the db field is of course still required
```
[the remark this answers is not in the document]
Comment (simen-rekkedal) [reply src/Core/Models/SystemUserUpdateDTO.cs:19]: the use case was not complete, we now know that only the IntegrationTitle will be updateable, the other fields will not be possible to change. Update is in the next delivery. Will remove the allownull, and add partyOrgNo

File: src/Authentication/Program.cs [hunk 8]
```diff
@@ -381,11 +378,10 @@ void ConfigureServices(IServiceCollection services, IConfiguration config)
         }
     });
 
-    services.AddAuthorization(options =>
-    {
-        options.AddPolicy(AuthzConstants.POLICY_SCOPE_SYSTEMREGISTER_WRITE, policy => policy
-            .RequireScopeAnyOf(AuthzConstants.SCOPE_SYSTEMREGISTER_ADMIN));
-    });
+    // Needed for the ResourceAccessHandler
```
[the remark this answers is not in the document]
Comment (simen-rekkedal) [reply src/Authentication/Program.cs:381]: ok

File: src/Authentication/Program.cs [hunk 9]
```diff
@@ -188,8 +185,8 @@ async Task ConnectToKeyVaultAndSetConfig(ConfigurationManager config)
         Environment.SetEnvironmentVariable("AZURE_CLIENT_SECRET", keyVaultSettings.ClientSecret);
         Environment.SetEnvironmentVariable("AZURE_TENANT_ID", keyVaultSettings.TenantId);
 
-        await SetUpAzureInsights(keyVaultSettings, config);        
-        AddAzureKeyVault(keyVaultSettings, config);
+        //await SetUpAzureInsights(keyVaultSettings, config);        
+        //AddAzureKeyVault(keyVaultSettings, config);
```
[the remark this answers is not in the document]
Comment (simen-rekkedal) [reply src/Authentication/Program.cs:189]: wrapped in if not dev

File: src/Authentication/Controllers/SystemUserController.cs [hunk 10]
```diff
@@ -119,15 +123,15 @@ public async Task<ActionResult> SetDeleteFlagOnSystemUser(Guid systemUserId)
...
-        [Authorize]
+        /// <returns></returns>    
+        //[Authorize]
         [Produces("application/json")]
         [ProducesResponseType(typeof(SystemUser), StatusCodes.Status200OK)]        
         [ProducesResponseType(StatusCodes.Status404NotFound)]        
-        [HttpPost("{partyId}")]
-        public async Task<ActionResult<SystemUser>> CreateSystemUser(int partyId, [FromBody] SystemUserRequestDto request)
+        [HttpPost("{partyOrgNo}")]
+        public async Task<ActionResult<SystemUser>> CreateSystemUser(string partyOrgNo, [FromBody] SystemUserRequestDto request)
```
[the remark this answers is not in the document]
Comment (simen-rekkedal) [reply src/Authentication/Controllers/SystemUserController.cs:132]: changed to reporteeOrgNo

**Review by acn-dgopa — APPROVED** [review]

**Pull request merged**